--- a/backport-include/linux/net.h
+++ b/backport-include/linux/net.h
@@ -2,13 +2,15 @@
 #define __BACKPORT_LINUX_NET_H
 #include_next <linux/net.h>
 #include <linux/static_key.h>
+#include <linux/version.h>
 
-
+#if LINUX_VERSION_IS_LESS(6,18,0)
 #ifndef SOCKWQ_ASYNC_NOSPACE
 #define SOCKWQ_ASYNC_NOSPACE   SOCK_ASYNC_NOSPACE
 #endif
 #ifndef SOCKWQ_ASYNC_WAITDATA
 #define SOCKWQ_ASYNC_WAITDATA   SOCK_ASYNC_WAITDATA
 #endif
+#endif /* LINUX_VERSION_IS_LESS(6,18,0) */
 
 #endif /* __BACKPORT_LINUX_NET_H */
