--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -5365,8 +5365,7 @@ static int mtk_get_rxfh(struct net_devic
 	if (!MTK_HAS_CAPS(eth->soc->caps, MTK_RSS) || !eth->soc->rss_num)
 		return -EOPNOTSUPP;
 
-	if (rxfh->hfunc)
-		rxfh->hfunc = ETH_RSS_HASH_TOP;	/* Toeplitz */
+	rxfh->hfunc = ETH_RSS_HASH_TOP;	/* Toeplitz */
 
 	if (rxfh->key)
 		memcpy(rxfh->key, rss_params->hash_key,
