--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -5168,6 +5168,22 @@ static void mtk_pending_work(struct work
 	rtnl_unlock();
 }
 
+static void mtk_napi_del(struct mtk_eth *eth)
+{
+	int i;
+
+	if (!eth->dummy_dev)
+		return;
+
+	netif_napi_del(&eth->tx_napi);
+	netif_napi_del(&eth->rx_napi[0].napi);
+
+	if (MTK_HAS_CAPS(eth->soc->caps, MTK_RSS)) {
+		for (i = 1; i < MTK_RX_RSS_NUM; i++)
+			netif_napi_del(&eth->rx_napi[MTK_RSS_RING(i)].napi);
+	}
+}
+
 static int mtk_free_dev(struct mtk_eth *eth)
 {
 	int i;
@@ -5193,12 +5209,15 @@ static int mtk_unreg_dev(struct mtk_eth
 
 	for (i = 0; i < MTK_MAX_DEVS; i++) {
 		struct mtk_mac *mac;
+
 		if (!eth->netdev[i])
 			continue;
+
 		mac = netdev_priv(eth->netdev[i]);
 		if (MTK_HAS_CAPS(eth->soc->caps, MTK_QDMA))
 			unregister_netdevice_notifier(&mac->device_notifier);
-		unregister_netdev(eth->netdev[i]);
+		if (eth->netdev[i]->reg_state == NETREG_REGISTERED)
+			unregister_netdev(eth->netdev[i]);
 	}
 
 	return 0;
@@ -6238,20 +6257,6 @@ static int mtk_probe(struct platform_dev
 		}
 	}
 
-	for (i = 0; i < MTK_MAX_DEVS; i++) {
-		if (!eth->netdev[i])
-			continue;
-
-		err = register_netdev(eth->netdev[i]);
-		if (err) {
-			dev_err(eth->dev, "error bringing up device\n");
-			goto err_deinit_ppe;
-		} else
-			netif_info(eth, probe, eth->netdev[i],
-				   "mediatek frame engine at 0x%08lx, irq %d\n",
-				   eth->netdev[i]->base_addr, eth->irq_fe[MTK_FE_IRQ_SHARED]);
-	}
-
 	/* we run 2 devices on the same DMA ring so we need a dummy device
 	 * for NAPI to work
 	 */
@@ -6259,9 +6264,10 @@ static int mtk_probe(struct platform_dev
 	if (!eth->dummy_dev) {
 		err = -ENOMEM;
 		dev_err(eth->dev, "failed to allocate dummy device\n");
-		goto err_unreg_netdev;
+		goto err_deinit_ppe;
 	}
-	netif_napi_add_weight(eth->dummy_dev, &eth->tx_napi, mtk_napi_tx, MTK_NAPI_WEIGHT);
+	netif_napi_add_weight(eth->dummy_dev, &eth->tx_napi, mtk_napi_tx,
+			      MTK_NAPI_WEIGHT);
 	netif_napi_add_weight(eth->dummy_dev, &eth->rx_napi[0].napi, mtk_napi_rx,
 			      MTK_NAPI_WEIGHT);
 
@@ -6272,6 +6278,21 @@ static int mtk_probe(struct platform_dev
 					      mtk_napi_rx, MTK_NAPI_WEIGHT);
 	}
 
+	for (i = 0; i < MTK_MAX_DEVS; i++) {
+		if (!eth->netdev[i])
+			continue;
+
+		err = register_netdev(eth->netdev[i]);
+		if (err) {
+			dev_err(eth->dev, "error bringing up device\n");
+			goto err_unreg_netdev;
+		}
+
+		netif_info(eth, probe, eth->netdev[i],
+			   "mediatek frame engine at 0x%08lx, irq %d\n",
+			   eth->netdev[i]->base_addr, eth->irq_fe[0]);
+	}
+
 	platform_set_drvdata(pdev, eth);
 	schedule_delayed_work(&eth->reset.monitor_work,
 			      MTK_DMA_MONITOR_TIMEOUT);
@@ -6280,6 +6301,9 @@ static int mtk_probe(struct platform_dev
 
 err_unreg_netdev:
 	mtk_unreg_dev(eth);
+	mtk_napi_del(eth);
+	free_netdev(eth->dummy_dev);
+	eth->dummy_dev = NULL;
 err_deinit_ppe:
 	mtk_ppe_deinit(eth);
 	mtk_mdio_cleanup(eth);
@@ -6313,13 +6337,7 @@ static void mtk_remove(struct platform_d
 	mtk_wed_exit();
 	mtk_hw_deinit(eth);
 
-	netif_napi_del(&eth->tx_napi);
-	netif_napi_del(&eth->rx_napi[0].napi);
-
-	if (MTK_HAS_CAPS(eth->soc->caps, MTK_RSS)) {
-		for (i = 1; i < MTK_RX_RSS_NUM; i++)
-			netif_napi_del(&eth->rx_napi[MTK_RSS_RING(i)].napi);
-	}
+	mtk_napi_del(eth);
 	mtk_cleanup(eth);
 	free_netdev(eth->dummy_dev);
 	mtk_mdio_cleanup(eth);
